crates/net/network/src/peers/manager.rs

```diff
@@ -298,7 +298,14 @@ impl PeersManager {
 
     /// Bans the peer temporarily with the configured ban timeout
     fn ban_peer(&mut self, peer_id: PeerId) {
-        self.ban_list.ban_peer_until(peer_id, std::time::Instant::now() + self.ban_duration);
+        let mut until = std::time::Instant::now() + self.ban_duration;
+        if let Some(peer) = self.peers.get(&peer_id) {
+            if peer.is_trusted() {
+                until = std::time::Instant::now() + self.backoff_durations.medium;
+            }
+        }
```

Comment:
this can be simplified a bit by tracking only the ban_duration as mut and constructing the until = Instant + banduration after the if check, this way we're not creating two instants.

this also needs an additional note on the function docs